File: app/src/main/java/com/example/irkpo_management/db/NoteRepository.java
package com.example.irkpo_management.db;

import android.content.Context;

public class NoteRepository {
    private final NoteDao noteDao;

    public NoteRepository(Context context) {
        noteDao = NoteDatabase.getInstance(context).noteDao();
    }

    // Конструктор для тестов
    NoteRepository(NoteDao noteDao) {
        this.noteDao = noteDao;
    }

    public void saveNote(String key, String text, Long remindAt) {
        noteDao.insert(new NoteEntity(key, text, remindAt));
    }

    public String loadNote(String key) {
        NoteEntity note = noteDao.getNoteForLesson(key);
        return note != null ? note.text : "";
    }

    public void deleteNote(String key) {
        noteDao.deleteByLessonKey(key);
    }

    public NoteEntity getNoteEntity(String key) {
        return noteDao.getNoteForLesson(key);
    }
}

File: app/src/main/java/com/example/irkpo_management/db/NoteRepository.kt
package com.example.irkpo_management.db

import android.content.Context

class NoteRepository {
    private val noteDao: NoteDao

    constructor(context: Context?) {
        noteDao = NoteDatabase.getInstance(context).noteDao()
    }

    // Конструктор для тестов
    internal constructor(noteDao: NoteDao) {
        this.noteDao = noteDao
    }

    fun saveNote(key: String?, text: String?, remindAt: Long?) {
        noteDao.insert(NoteEntity(key, text, remindAt))
    }

    fun loadNote(key: String?): String? {
        val note = noteDao.getNoteForLesson(key)
        return if (note != null) note.text else ""
    }

    fun deleteNote(key: String?) {
        noteDao.deleteByLessonKey(key)
    }

    fun getNoteEntity(key: String?): NoteEntity? {
        return noteDao.getNoteForLesson(key)
    }
}
